### PR TITLE
chore(docs): Update overview.mdx

### PR DESCRIPTION
**Pull request by laurapacilio**
A few tiny updates to the new API reference overview page :)

### CODE REVIEW

File: website/docs/cdktf/api-reference/overview.mdx [hunk 1]
```diff
@@ -1,17 +1,17 @@
 ---
 page_title: API Reference
```
Comment (laurapacilio) [inline website/docs/cdktf/api-reference/overview.mdx:2]:
```suggestion
page_title: API Reference - CDKTF Core Library
```

File: website/docs/cdktf/api-reference/overview.mdx [hunk 2]
```diff
@@ -1,17 +1,17 @@
 ---
-page_title: API Reference
+page_title: API Reference - CDKTF Core Library
 description: >-
-  The CDKTF core library is what you use to specify your CDKTF program alongside with the generated provider bindings.
+  Use the CDKTF core library with generated provider bindings to write your CDKTF program. Get API reference in TypeScript, Python, Java, CSharp, and Go.
```
Comment (laurapacilio) [inline website/docs/cdktf/api-reference/overview.mdx:4]:
```suggestion
  Use the CDKTF core library with generated provider bindings to write your CDKTF program. It is available in TypeScript, Python, Java, CSharp, and Go.
```

File: website/docs/cdktf/api-reference/overview.mdx [hunk 3]
```diff
@@ -1,17 +1,17 @@
 ---
-page_title: API Reference
+page_title: API Reference - CDKTF Core Library
```
Comment (laurapacilio) [inline website/docs/cdktf/api-reference/overview.mdx:2]: I was thinking maybe we should be specific that this is for the core CDK library in the title and not providers, etc. Do you think this makes sense to do, or you think it's confusing?

**Review by DanielMSchmidt — APPROVED** [review]

File: website/docs/cdktf/api-reference/overview.mdx (hunk 3 above)
Comment (DanielMSchmidt) [reply website/docs/cdktf/api-reference/overview.mdx:2]: Sounds good 👍

**Comment by laurapacilio** [comment]
@DanielMSchmidt could you merge this? I still don't have permissions :)

**Pull request merged**